### ISSUE TITLE
Brand color picker should suggest accessible combinations

### ISSUE THREAD

**Issue by marcysutton** [forem/forem#11504]
## How to ensure user-configurable colors in Forem meet contrast ratio requirements

Text, links, and other interactive controls (like icon buttons) need to meet the minimum color contrast requirements laid out in [WCAG 1.4.3](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html): 4.5:1 for regular text and 3:1 for bold or large text (18pt/24px). When users can pick a brand color, Forem should have a system for ensuring that color is accessible – including any secondary colors and usage.

Related issues:

- https://github.com/forem/forem/issues/735
- https://github.com/forem/forem/issues/1273
- https://github.com/forem/forem/issues/1774
- https://github.com/forem/forem/issues/2152

**Describe the solution you'd like**

Create a UI for users to pick an accessible color combination, using CSS and JavaScript functions to suggest color palettes based on a single brand/accent color. For example:
- A user could pick any color, and:
- A UI would be presented to review an accessible color palette
- A small number of pre-made, accessible color palettes could also be surfaced for users to choose from.

Adding this functionality would require currently-set brand colors to be revisited so they work with a new system, perhaps by prompting the user to double-check their theme settings?

**Describe alternatives you've considered**

- Not doing anything would mean color combos would never be validated for accessibility unless the user specifically tested their own color values, which is unlikely.
- Letting users pick two brand colors and then trying to make them accessible in all use cases could be much more complicated (more variables and less likelihood the colors will pass at all)

**Additional context / Prior art**

- Letting users choose any color and automatically create a palette using contrast rules: https://medium.com/confrere/automatically-creating-an-accessible-color-palette-from-any-color-sure-e735c3f2f45e 
- JS library for immutable color conversion and manipulation with support for CSS color strings (recommended in Confrere Medium piece): https://www.npmjs.com/package/color
- Designing accessible color systems: https://stripe.com/blog/accessible-color-systems
- Color suggestions module for specific contrast ratios: https://github.com/A11yance/a11y-color
- Accessible color palette tool based on 18F project: https://github.com/toolness/accessible-color-matrix 
- JS Color picker used in 18F project: https://jscolor.com/

**Comment by rhymes** [comment]
@marcysutton adding this to the mix as well https://github.com/forem/forem/issues/11719#issue-755739942
